src/ripple/json/json_conversions.h

```diff
@@ -0,0 +1,71 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_JSON_JSON_CONVERSIONS_H_INCLUDED
+#define RIPPLE_JSON_JSON_CONVERSIONS_H_INCLUDED
+
+#include <ripple/basics/safe_cast.h>
+#include <ripple/basics/tagged_integer.h>
+
+namespace ripple
+{
+// Forward declaration - defined in ReadView.h
+struct DropsTag;
+}
+
+namespace Json
+{
+
+namespace detail
+{
+
+template <class Integer>
+UInt
+maybe_toUInt(ripple::tagged_integer<Integer, ripple::DropsTag> drops,
+    std::false_type)
+{
+    using namespace ripple;
+    return unsafe_cast<UInt>(drops.value());
+}
+
+template <class Integer>
+UInt
+maybe_toUInt(ripple::tagged_integer<Integer, ripple::DropsTag> drops,
+    std::true_type)
+{
+    using namespace ripple;
+    return safe_cast<UInt>(drops.value());
+}
+
+}
+
+template<class Integer,
+    class = typename std::enable_if<
+        std::is_integral<Integer>::value >::type>
+UInt
+toUInt(ripple::tagged_integer<Integer, ripple::DropsTag> drops)
```

Comment:
Because the API only ever shows drops to the user (unless I missed a spot), so there's no need to generalize. If someone tries call the function with a fee unit value, they'll get a compiler error and know something is wrong.